habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py

```diff
@@ -71,29 +71,43 @@ def __init__(
             lr_schedule_fn,
         )
 
+        num_active_agents = sum(self._pop_config.num_active_agents_per_type)
+        (
+            self._multi_policy,
+            self._multi_updater,
+            self._multi_storage,
+        ) = self._create_multi_components(config, env_spec, num_active_agents)
+
+        if self.nbuffers != 1:
+            raise ValueError(
+                "Multi-agent training does not support double buffered sampling"
+            )
+        if config.habitat_baselines.evaluate:
+            self._sample_active()
+
+    def _create_multi_components(self, config, env_spec, num_active_agents):
         if self._pop_config.self_play_batched:
             policy_cls: Type = SelfBatchedPolicy
             updater_cls: Type = SelfBatchedUpdater
             storage_cls: Type = SelfBatchedStorage
-            self._active_agents = [0, 0]
+            self._active_agents = np.array([0, 0])
         else:
             policy_cls = MultiPolicy
             updater_cls = MultiUpdater
             storage_cls = MultiStorage
 
         # TODO(xavi to andrew): why do we call these functions? It seems they
         # just create an empty class
-        num_active_agents = sum(self._pop_config.num_active_agents_per_type)
```

Comment:
Same comment as before, it seems like these classes do not initialize the arguments you are passing.